l402/header.go

```diff
@@ -134,10 +142,17 @@ func SetHeader(header *http.Header, mac *macaroon.Macaroon,
 	if err != nil {
 		return err
 	}
-	value := fmt.Sprintf(
-		authFormat, base64.StdEncoding.EncodeToString(macBytes),
-		preimage.String(),
-	)
-	header.Set(HeaderAuthorization, value)
+	macStr := base64.StdEncoding.EncodeToString(macBytes)
+	preimageStr := preimage.String()
+
+	// Send "Authorization: LSAT..." header before sending
+	// "Authorization: L402" header to be compatible with old aperture.
+	// TODO: remove this after aperture is upgraded everywhere.
```

Comment:
We can't really remove this when we upgraded _aperture_ everywhere. Only once there is no old client out there anymore (which can take years) can we remove this. So in practice it will probably remain for a long time (which is okay IMO).

Reply:
This function is called on the client side. After all servers are upgraded, this backward compatibility layer of clients can be removed, IIUC.

Old clients will continue to send "Authorization: LSAT" and new aperture will accept it even if this code is removed. This is provided in another [server-side] function `FromHeader` which will have to keep backward compatibility for years.

Reply:
Ah, you're right, this is client side only. I guess I got "WWW-Authenticate" and "Authorization" mixed up, seems like mental context reload was insufficient :sweat_smile: